src/spring/azext_spring/custom.py

```diff
@@ -110,15 +112,24 @@ def spring_update(cmd, client, resource_group, name, app_insights_key=None, app_
     updated_resource_properties.zone_redundant = None
 
     if enable_log_stream_public_endpoint is not None or enable_dataplane_public_endpoint is not None:
+        if updated_resource_properties.vnet_addons is None:
+            updated_resource_properties.vnet_addons = models.ServiceVNetAddons()
         val = enable_log_stream_public_endpoint if enable_log_stream_public_endpoint is not None else \
             enable_dataplane_public_endpoint
-        updated_resource_properties.vnet_addons = models.ServiceVNetAddons(
-            data_plane_public_endpoint=val,
-            log_stream_public_endpoint=val
-        )
+        updated_resource_properties.vnet_addons.data_plane_public_endpoint=val
+        updated_resource_properties.vnet_addons.log_stream_public_endpoint=val
         update_dataplane_public_endpoint = True
-    else:
-        updated_resource_properties.vnet_addons = None
+
+    if enable_private_storage_access is not None:
+        if updated_resource_properties.vnet_addons is None:
+            updated_resource_properties.vnet_addons = models.ServiceVNetAddons(
+                # explicitly set as none in case unexpected update
+                data_plane_public_endpoint=None,
```

Comment:
Is `None` the same as `null`, or more specifically, does `None` has the effect of "do not change" rather than "disable"?